cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -683,9 +696,21 @@ private BatchPhaseSpec getPhaseSpec(String programName, PipelinePhase phase) {
       }
     }
 
+    // overide spec with runtime arguments
+    boolean processTimingEnabled = spec.isProcessTimingEnabled();
+    Map<String, String> properties = spec.getProperties();
+    try {
+      Map<String, String> runtimeArguments = applicationConfigurer.getRuntimeConfigurer().getRuntimeArguments();
```

Comment:
this runtime configurer will be null for initial deploy, you will need to handle null here